chore: Merges Azure KMS Encryption at Rest Private Endpoint feature to master (#2569)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/encryption_at_rest.md

# Data Source: mongodbatlas_encryption_at_rest

`mongodbatlas_encryption_at_rest` describes encryption at rest configuration for an Atlas project with one of the following providers:

[Amazon Web Services Key Management Service](https://docs.atlas.mongodb.com/security-aws-kms/#security-aws-kms)
[Azure Key Vault](https://docs.atlas.mongodb.com/security-azure-kms/#security-azure-kms)
[Google Cloud KMS](https://docs.atlas.mongodb.com/security-gcp-kms/#security-gcp-kms)

> ~> **IMPORTANT** Atlas encrypts all cluster storage and snapshot volumes, securing all cluster data on disk: a concept known as encryption at rest, by default.

Reviewer: ```suggestion
~> **IMPORTANT** By default, Atlas enables encryption at rest for all cluster storage and snapshot volumes.
```

Author: updated this in the PR
